examples/mne_util.py

```diff
@@ -0,0 +1,294 @@
+                                        subjects_dir=subjects_dir)
+        elif isinstance(l, mne.BiHemiLabel):
+            l.lh.values.fill(1.0)
+            l.rh.values.fill(1.0)
+            labels_morphed[i].lh = l.lh.morph(subject_to=subject_to,
+                                              subjects_dir=subjects_dir)
+            labels_morphed[i].rh = l.rh.morph(subject_to=subject_to,
+                                              subjects_dir=subjects_dir)
+
+    # make sure there are no duplicate labels
+    labels_morphed = sorted(list(set(labels_morphed)), key=lambda x: x.name)
+
+    return labels_morphed
+
+
+def apply_projs(epochs, fwd, cov):
+    """ apply projection operators to fwd and cov """
+    proj, _ = mne.io.proj.setup_proj(epochs.info, activate=False)
+    G = fwd['sol']['data']
+    fwd['sol']['data'] = np.dot(proj, G)
+
+    Q = cov.data
+    if not np.allclose(np.dot(proj, Q), Q):
+        Q = np.dot(proj, np.dot(Q, proj.T))
+        cov.data = Q
+
+    return fwd, cov
+
+
+def scale_sensor_data(epochs, fwd, cov, roi_to_src, eeg_scale=1., mag_scale=1.,
+    grad_scale=1.):
+    """ apply per-channel-type scaling to epochs, forward, and covariance """
+    # from util import Carray ##skip import just pasted; util also from MEGLDS repo
+    Carray64 = lambda X: np.require(X, dtype=np.float64, requirements='C')
+    Carray32 = lambda X: np.require(X, dtype=np.float32, requirements='C')
+    Carray = Carray64
+
+    # get indices for each channel type
+    ch_names = cov['names']  # same as self.fwd['info']['ch_names']
+    sel_eeg = pick_types(fwd['info'], meg=False, eeg=True, ref_meg=False)
+    sel_mag = pick_types(fwd['info'], meg='mag', eeg=False, ref_meg=False)
+    sel_grad = pick_types(fwd['info'], meg='grad', eeg=False, ref_meg=False)
+    #2 channels are removed so idx != ch_name   
+    #can we do idx = c for c in sel??
+    #idx_eeg = [ch_names.index(ch_names[c]) for c in sel_eeg]
+    #idx_mag = [ch_names.index(ch_names[c]) for c in sel_mag]
+    #idx_grad = [ch_names.index(ch_names[c]) for c in sel_grad]
+    idx_eeg = [c for c in sel_eeg]
+    idx_mag = [c for c in sel_mag]
+    idx_grad = [c for c in sel_grad]
+
+    # retrieve forward and sensor covariance
+    G = fwd['sol']['data'].copy()
+    Q = cov.data.copy()
+
+    # scale forward matrix
+    G[idx_eeg,:] *= eeg_scale
+    G[idx_mag,:] *= mag_scale
+    G[idx_grad,:] *= grad_scale
+
+    # construct GL matrix
+    GL = Carray(csr_matrix.dot(roi_to_src.L.T, G.T).T)
+
+    # scale sensor covariance
+    Q[np.ix_(idx_eeg, idx_eeg)] *= eeg_scale**2
+    Q[np.ix_(idx_mag, idx_mag)] *= mag_scale**2
+    Q[np.ix_(idx_grad, idx_grad)] *= grad_scale**2
+
+    # scale epochs
+    info = epochs.info.copy()
+    data = epochs.get_data().copy()
+
+    data[:,idx_eeg,:] *= eeg_scale
+    data[:,idx_mag,:] *= mag_scale
+    data[:,idx_grad,:] *= grad_scale
+
+    epochs = mne.EpochsArray(data, info)
+
+    return G, GL, Q, epochs
```

Comment:
With `scales = 1` the `scale_sensor_data()` is not necessary. Should we leave it as part of LDS in case users want to change the scales? 

Also, I'm not following how the steps in `scale_sensor_data()` is the same as the whitening functions you've mentioned. Can you checkout my previous push with commit `does scale_sensor_data() == whitening?` for lines 64-72 in `state_space_connectivity.py()` to see if I'm following you correctly?